src/Interpreters/InterpreterShowCreateQuery.cpp

```diff
@@ -94,10 +95,9 @@ QueryPipeline InterpreterShowCreateQuery::executeImpl()
         create.to_inner_uuid = UUIDHelpers::Nil;
     }
 
-    String res = create_query->formatWithSecretsHidden(/* max_length= */ 0, /* one_line= */ false);
-
     MutableColumnPtr column = ColumnString::create();
-    column->insert(res);
+    SecretHidingFormatSettings tmp{.ctx = *getContext(), .query = *create_query, .one_line = false};
```

Comment:
Strange thing here: moving out struct initializer to separate variable (rvalue->lvalue ref) changes the behavior to correct.